src/nn/wasm/PreparedModel.js

```diff
@@ -1265,8 +1265,14 @@ export default class PreparedModel {
         OPS_CHECK(output.runtimeshape.DimensionsCount() === perm.length);
 
         // Extend perm to length 4 by appending 0
-        for (let i = perm.length; i < 4; ++i) {
-          perm.push(0);
+        if (perm instanceof Array) {        
```

Comment:
Is `perm` always a typed array? If it is, please remove this block and just keep the `else` block. Thanks!

Reply:
Not always, because according the [code](https://github.com/intel/webml-polyfill/blob/f7edb9685f06fc8ddcb4e48d436690bcc786f580/src/nn/wasm/PreparedModel.js#L1250-L1257) here the default perm is an Array if the operation has only one operand. My first PR is trying to convert typedArray to Array, I almost forgot that.

Reply:
Thanks for the clarification. Looks good to me.